test/end2end_test.go

```diff
@@ -7676,7 +7676,14 @@ func (s) TestDeadlineSetOnConnectionOnClientCredentialHandshake(t *testing.T) {
 	if err != nil {
 		t.Fatalf("Failed to listen: %v", err)
 	}
-	go lis.Accept()
+	go func() {
+		conn, err := lis.Accept()
+		if err != nil {
+			t.Errorf("Error accepting connection: %v", err)
+			return
+		}
+		defer conn.Close()
```

Comment:
This will close the conn immediately since it's the last line of the function.

This _should_ be fine (but remove `defer`) and I don't _think_ it's racy, but you should probably run a bunch of iterations to be sure.

The "ideal" way to do this, though, would be to put the conn on a channel and defer close in the main goroutine:

```go
connCh := make(chan net.Conn, 1)
go func() {
		defer close(connCh)
		conn, err := lis.Accept()
		if err != nil {
			t.Errorf("Error accepting connection: %v", err)
			return
		}
		connCh<-conn
}()
defer func() {
	conn := <-connCh
	if conn != nil {
		conn.Close()
	}
}()
```

Reply:
Okay, this was based on what was in the testing framework for e2e tests. Changed to your way.

Reply:
> Okay, this was based on what was in the testing framework for e2e tests. Changed to your way.

Never blindly trust someone else's code (especially mine).  :)